node/src/lib.rs

```diff
@@ -341,8 +347,12 @@ impl Node {
                 store.batch_store.clone(),
                 metrics.clone(),
             );
-            handles.extend(worker_handles);
+            // TODO: propagate worker task names if needed.
```

Comment:
should this be an issue?

Reply:
Opened an issue and linked here. If other aspects of this PR look ok, I can make the change in this PR too.